lib/taskana-core-test/src/test/java/acceptance/task/create/CreateTaskWithAttachmentAccTest.java

```diff
@@ -0,0 +1,201 @@
+    taskToCreate.setClassificationKey(defaultClassificationSummary.getKey());
+    taskToCreate.setPrimaryObjRef(defaultObjectReference);
+    taskToCreate.addAttachment(copiedAttachment);
+    Task result = taskService.createTask(taskToCreate);
+
+    assertThat(result.getAttachments().get(0).getTaskId()).isEqualTo(result.getId());
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_FetchAttachmentClassification_When_CreatingTaskWithAttachments() throws Exception {
+    Attachment attachment = taskService.newAttachment();
+    attachment.setObjectReference(defaultObjectReference);
+    ClassificationSummary classification =
+        classificationService
+            .newClassification(
+                defaultClassificationSummary.getKey(),
+                defaultClassificationSummary.getDomain(),
+                defaultClassificationSummary.getType())
+            .asSummary();
+    attachment.setClassificationSummary(classification);
+    Task taskToCreate = taskService.newTask(defaultWorkbasketSummary.getId());
+    taskToCreate.setClassificationKey(defaultClassificationSummary.getKey());
+    taskToCreate.setPrimaryObjRef(defaultObjectReference);
+    taskToCreate.addAttachment(attachment);
+    taskToCreate = taskService.createTask(taskToCreate);
+
+    classification = taskToCreate.getAttachments().get(0).getClassificationSummary();
+    assertThat(classification).isEqualTo(defaultClassificationSummary);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @TestFactory
+  Stream<DynamicTest>
+      should_ThrowException_When_CreatingTaskWithInvalidObjectReferenceOfAttachment() {
+    ObjectReference objRefTypeNull =
+        taskService.newObjectReference("Company", "System", "Instance", null, "Value");
+    ObjectReference objRefValueNull =
+        taskService.newObjectReference("Company", "System", "Instance", "Type", null);
+    ObjectReference objRefCompanyNull =
+        taskService.newObjectReference(null, "System", "Instance", "Type", "Value");
+    List<Pair<String, ObjectReference>> valuesForTests =
+        List.of(
+            Pair.of("ObjRef is null", null),
+            Pair.of("Type of objRef is null", objRefTypeNull),
+            Pair.of("Value of objRef is null", objRefValueNull),
+            Pair.of("Company of objRef is null", objRefCompanyNull));
+
+    ThrowingConsumer<Pair<String, ObjectReference>> test =
+        p -> {
+          Task task = taskService.newTask(defaultWorkbasketSummary.getId());
+          task.setClassificationKey(defaultClassificationSummary.getKey());
+          task.setPrimaryObjRef(defaultObjectReference);
+          Attachment attachment = taskService.newAttachment();
+          attachment.setClassificationSummary(defaultClassificationSummary);
+          attachment.setObjectReference(p.getRight());
+          task.addAttachment(attachment);
+          assertThatThrownBy(() -> taskService.createTask(task))
+              .isInstanceOf(InvalidArgumentException.class);
```

Comment:
I know that the original tests did not verify the message of the IllegalArgumentException, but I am undecided whether we should verify the message of those Exceptions. What do you think?

Reply:
I will verify the message